Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -211,6 +222,16 @@ void PaintableBox::paint_background(PaintContext& context) const
     if (computed_values().border_top().width || computed_values().border_right().width || computed_values().border_bottom().width || computed_values().border_left().width)
         background_rect = absolute_border_box_rect();
 
+    if (computed_values().clip().is_rect() && computed_values().position() == CSS::Position::Absolute) {
+        auto clip_rect = Gfx::FloatRect {
+            background_rect.x() + computed_values().clip().to_rect().left_edge.to_px(Paintable::layout_node()),
+            background_rect.y() + computed_values().clip().to_rect().top_edge.to_px(Paintable::layout_node()),
+            computed_values().clip().to_rect().right_edge.to_px(Paintable::layout_node()) - computed_values().clip().to_rect().left_edge.to_px(Paintable::layout_node()),
+            computed_values().clip().to_rect().bottom_edge.to_px(Paintable::layout_node()) - computed_values().clip().to_rect().top_edge.to_px(Paintable::layout_node())
```

Comment:
This would then be `clip_rect.resolved().translated_by(background_rect.location()`

Reply:
Sweet, yeah that's way better